test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1070,6 +1070,9 @@ private synchronized void reset(boolean wipeData) throws IOException {
             wipePendingDataDirectories();
         }
 
+        assert nodes.isEmpty() || nodes.values().stream().anyMatch(NodeAndClient::isMasterEligible)
```

Comment:
can you use `assertTrue` here so that this does not only fail when assertions are enabled?